test/v200/utils/common/command_test_utils.go

```diff
@@ -0,0 +1,212 @@
+		AttributeName := "Name_" + GetRandomString(6, false)
+		attributes[AttributeName] = "Value_" + GetRandomString(6, false)
+		LogInfoMessage(fmt.Sprintf("Add attribute : %s = %s", AttributeName, attributes[AttributeName]))
+	}
+	return attributes
+}
+
+// addGroup creates and returns a group in a schema structure
+func (testDevFile *TestDevfile) addGroup() *schema.CommandGroup {
+
+	commandGroup := schema.CommandGroup{}
+	commandGroup.Kind = GetRandomGroupKind()
+	LogInfoMessage(fmt.Sprintf("group Kind: %s, default already set %t", commandGroup.Kind, testDevFile.GroupDefaults[commandGroup.Kind]))
+	// Ensure only one and at least one of each type are labelled as default
+	if !testDevFile.GroupDefaults[commandGroup.Kind] {
+		testDevFile.GroupDefaults[commandGroup.Kind] = true
+		commandGroup.IsDefault = true
+	} else {
+		commandGroup.IsDefault = false
+	}
+	LogInfoMessage(fmt.Sprintf("group isDefault: %t", commandGroup.IsDefault))
+	return &commandGroup
+}
+
+// AddCommand creates a command of a specified type in a schema structure and pupulates it with random attributes
+func (testDevFile *TestDevfile) AddCommand(commandType schema.CommandType) schema.Command {
+
+	var command *schema.Command
+	if commandType == schema.ExecCommandType {
+		command = testDevFile.createExecCommand()
+		testDevFile.SetExecCommandValues(command)
+	} else if commandType == schema.CompositeCommandType {
+		command = testDevFile.createCompositeCommand()
+		testDevFile.SetCompositeCommandValues(command)
+	} else if commandType == schema.ApplyCommandType {
+		command = testDevFile.createApplyCommand()
+		testDevFile.SetApplyCommandValues(command)
+	}
+	return *command
+}
+
+// createExecCommand creates and returns an empty exec command in a schema structure
+func (testDevFile *TestDevfile) createExecCommand() *schema.Command {
+
+	LogInfoMessage("Create an exec command :")
+	command := schema.Command{}
+	command.Id = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("command Id: %s", command.Id))
+	command.Exec = &schema.ExecCommand{}
+	testDevFile.commandAdded(command)
+	return &command
+
+}
+
+// SetExecCommandValues randomly sets exec command attribute to random values
+func (testDevFile *TestDevfile) SetExecCommandValues(command *schema.Command) {
+
+	execCommand := command.Exec
+
+	// exec command must be mentioned by a container component
+	execCommand.Component = testDevFile.GetContainerName()
+
+	execCommand.CommandLine = GetRandomString(4, false) + " " + GetRandomString(4, false)
+	LogInfoMessage(fmt.Sprintf("....... commandLine: %s", execCommand.CommandLine))
+
+	// If group already leave it to make sure defaults are not deleted or added
+	if execCommand.Group == nil {
+		if GetRandomDecision(2, 1) {
+			execCommand.Group = testDevFile.addGroup()
+		}
+	}
+
+	if GetBinaryDecision() {
+		execCommand.Label = GetRandomString(12, false)
+		LogInfoMessage(fmt.Sprintf("....... label: %s", execCommand.Label))
+	} else {
+		execCommand.Label = ""
```

Comment:
no need to set this Else statement. the fields are defined as `omitempty`, will be set to `""` or `nil` if not present. 
same as the other else statements in  this function

Reply:
This function is also used to update an existing exec command where, for example, the label may already be set.

Reply:
got it. And updating the description would help better understanding, thanks.

Reply:
Fixed in new commit